observability/test/run_o11y_tests.py

```diff
@@ -0,0 +1,1174 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Run one gRPC Observability integration tests test case"""
+
+import argparse
+import base64
+from datetime import datetime, timezone
+from enum import Enum
+from google.cloud import (
+    logging_v2,
+    monitoring_v3,
+    trace_v1,
+)
+from google.cloud.logging_v2.types import LogEntry
+from google.cloud.monitoring_v3.types.metric_service import ListTimeSeriesResponse
+from google.cloud.trace_v1.services.trace_service.pagers import ListTracesPager
+from google.cloud.trace_v1.types import ListTracesResponse
+from google.protobuf.timestamp_pb2 import Timestamp
+import json
+from kubernetes import ( # type: ignore
+    client as kubernetes_client,
+    config as kubernetes_config,
+    utils as kubernetes_utils,
+)
+from kubernetes.stream import stream as kubernetes_stream # type: ignore
+import logging
+import os
+import random
+import re
+import signal
+import string
+import subprocess
+import sys
+from test_utils import (
+    ClientActionArgs,
+    ConfigLocation,
+    ExpectCount,
+    JobMode,
+    LanguageConstants,
+    LoggerSide,
+    ObservabilityConfig,
+    ObservabilityTestCase,
+    ServerStartArgs,
+    SupportedLang,
+    SupportedLangEnum,
+    TestRunMetadata,
+    TestUtil,
+)
+import time
+import traceback
+from typing import Any, Dict, List, Optional, TextIO, Tuple, TypeVar, Type
+import unittest
+
+PROJECT = 'microsvcs-testing'
+PROJECT_NUM = '168376032566'
+CLUSTER_NAME = 'grpc-o11y-integration-testing-cluster'
+ZONE = 'us-central1-c'
+OBSERVABILITY_LOG_NAME = 'microservices.googleapis.com%2Fobservability%2Fgrpc'
+CONFIG_ENV_VAR_NAME = 'GRPC_GCP_OBSERVABILITY_CONFIG'
+CONFIG_FILE_ENV_VAR_NAME = 'GRPC_GCP_OBSERVABILITY_CONFIG_FILE'
+IMAGE_TAG = '1.53.0-dev'
+SUPPORTED_METRICS = [
+    'custom.googleapis.com/opencensus/grpc.io/client/started_rpcs',
+    'custom.googleapis.com/opencensus/grpc.io/server/started_rpcs',
+    'custom.googleapis.com/opencensus/grpc.io/client/completed_rpcs',
+    'custom.googleapis.com/opencensus/grpc.io/server/completed_rpcs',
+]
+# number of seconds to allow running RPC action command
+WAIT_SECS_CLIENT_ACTION = 95
+WAIT_SECS_SERVER_START = 20
+WAIT_SECS_GKE_DEPLOYMENT = 150
+# wait for this many seconds after RPC action to allow exporters to flush
+WAIT_SECS_EXPORTER_FLUSH = 65
+WAIT_SECS_READY = 20
+
+LANGUAGE_CONSTANTS = {
+    SupportedLangEnum.GO: LanguageConstants(
+        repo_name = 'grpc-go',
+        server_start_cmd = 'interop/observability/server/server',
```

Comment:
Please let's not have custom configuration for each. We should define the arguments including choosing between client/server and all support that specific set.

Reply:
Yes I plan to eventually remove this and make them parameters (or just a uniform naming scheme somewhere). I will either add a TODO (more likely) or fix it by the next revision (less likely).

Reply:
This whole `LANGUAGE_CONSTANTS` block has been removed! Now these details live in individual language's `Dockerfile` and `run.sh`